gcivil-nyu-org/fall2019-cs-gy-6063-team-debug_entities
# Revamped logic and models for matching

### Description

I changed the models and logic involved in making matches, so now it should be simpler and faster and more robust.

### Testing Directions

```
python manage.py makemigrations
python manage.py migrate
python manage.py pull_seatgeek_data
```

1. Get the code from this PR
1. Run the Django server
1. Make 3 users. One of them will be you (a superuser) and the other 2 will be people you swipe on. I'll call these other users U1 and U2. Make all 3 people be interested in the same event. In my examples, the event will be Rosanne Cash. The admin interface is probably the easiest way to do this.
1. In the admin interface, give U1 a swipe on you to the right. In my case, "you" is vm1564@nyu.edu. If "Direction" is checked, that means it was a swipe to the right.
![image](https://user-images.githubusercontent.com/13837978/68074040-e44c1f00-fd6c-11e9-973d-1659f132bd40.png)
1. Now go to the matching stack for the event. Our goal is to see both U1 and U2 in the stack and if we swipe right on both, we will match with U1 and not with U2 because only U1 swiped right on us.
1. See U1 in the stack, as desired.
![image](https://user-images.githubusercontent.com/13837978/68074135-422d3680-fd6e-11e9-9ae1-47b41427db8a.png)
1. Click the thumbs up and see in the console that you matched, as desired. The values in the console output are the user IDs, not their names. My ID is 1, U1's ID is 2.
![image](https://user-images.githubusercontent.com/13837978/68074318-18750f00-fd70-11e9-84be-2fae52ad6f16.png)
1. Notice that after clicking the thumbs up on U1, you still see U1. That's a bug. Click the thumbs up again and you'll see U2. I don't know why you have to click twice but we'll fix it later.
1. Click the thumbs up on U2 and see that the console does not say that you matched, as desired.

### Miscellaneous Comments

* These tests are not exhaustive. There are other scenarios, such as checking that someone who swiped left on you does not show up in your stack. I just couldn't write every single possibility in the instructions. Test any scenarios you can think of.
* A nice unintended consequence of these changes is that test coverage increased a lot just because I deleted a lot of code.

### Checklist
